File: apps/web/src/extraction/navigatum.py
import requests

NAVIGATUM_URL = "https://nav.tum.de"

def get_base_infos(room_id: str):
    """Returns the campus/building info for a given room ID."""
    url = f"{NAVIGATUM_URL}/api/locations/{room_id}"
    response = requests.get(url, params={"lang": "en"})
    response.raise_for_status()
    data = response.json()

    return {
        "room": data.get("name"),
        "parents": data.get("parents"),        # list of parent IDs
        "parent_names": data.get("parent_names"),  # list of parent names (readable)
        "type": data.get("type"),
    }

def get_room (room_id: str):
    base_infos = get_base_infos(room_id)
    return base_infos["room"]

def get_campus(room_id: str):
    base_infos = get_base_infos(room_id)
    return base_infos['parent_names'][1]

File: apps/web/src/extraction/navigatum.ts
const NAVIGATUM_URL = "https://nav.tum.de";

interface RoomBaseInfos {
  room: string;
  parents: string[];
  parent_names: string[];
  type: string;
}

export async function getBaseInfos(roomId: string): Promise<RoomBaseInfos> {
  const res = await fetch(`${NAVIGATUM_URL}/api/locations/${roomId}?lang=en`);
  if (!res.ok) throw new Error(`Failed to fetch room info for ${roomId}: ${res.status}`);
  const data = await res.json();
  return {
    room: data.name,
    parents: data.parents,
    parent_names: data.parent_names,
    type: data.type,
  };
}

export async function getRoom(roomId: string): Promise<string> {
  const info = await getBaseInfos(roomId);
  return info.room;
}

export async function getCampus(roomId: string): Promise<string> {
  const info = await getBaseInfos(roomId);
  return info.parent_names[1] ?? "";
}
